one-offs/migrate-boms/migrate-boms.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"log"
+	"regexp"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/option"
+)
+
+var (
+	srcBucket   = flag.String("srcBucket", "halconfig", "The GCS bucket name to read from. Must contain a bom/ directory.")
+	destBucket  = flag.String("destBucket", "halconfig2", "The GCS bucket name to write to.")
+	jsonKeyPath = flag.String("jsonKey", "", "Filepath to JSON key with permission to read --src-bucket and write to the --dest-bucket")
+
+	releaseRegexp = regexp.MustCompile(`1\.[0-9]{1,2}\.[0-9]{1,2}\.yml`)
+)
+
+func main() {
+	flag.Parse()
+
+	ctx := context.Background()
+	storageSvc, err := storage.NewClient(ctx, option.WithCredentialsFile(*jsonKeyPath), option.WithScopes(storage.ScopeFullControl))
+	if err != nil {
+		log.Fatalf("Error generating new storage client: %v", err)
+	}
+
+	iter := storageSvc.Bucket(*srcBucket).Objects(ctx, &storage.Query{Prefix: "bom/"})
+	for obj, err := iter.Next(); err == nil; obj, err = iter.Next() {
+		if releaseRegexp.Match([]byte(obj.Name)) {
+			log.Printf("Reading %v\n", obj.Name)
+			r, err := storageSvc.Bucket(*srcBucket).Object(obj.Name).NewReader(ctx)
+			if err != nil {
+				log.Fatalf("error reading object %v: %v", obj.Name, err)
+			}
+			scanner := bufio.NewScanner(r)
+			w := storageSvc.Bucket(*destBucket).Object(obj.Name).NewWriter(ctx)
+			var replaced bool
+			for scanner.Scan() {
+				t := scanner.Text()
+				if strings.Contains(t, "gcr.io/spinnaker-marketplace") {
+					t = strings.ReplaceAll(t, "gcr.io/spinnaker-marketplace", "us-docker.pkg.dev/spinnaker-community/releases")
+					replaced = true
+					log.Printf("Replaced! New line: %v", t)
+				}
+				_, err := w.Write([]byte(t + "\n"))
+				if err != nil {
+					log.Fatalf("error writing output: %v", err)
+				}
+			}
+			if replaced {
```

Comment:
I'm not sure I understand why we're only writing to the destination bucket if we replaced a string.  Wouldn't we want to move the BOM over to the new bucket anyway?  Or are we assuming that this wasn't actually a BOM if we didn't find a reference to the repository in it?

Reply:
(And if we do want to skip writing in some cases, we should probably create a loop-scoped context and cancel it in the event we choose to abandon the write-in-progress as I think we're leaving a lot of dangling writes here.)

Reply:
The "production" run intends to have srcBucket == destBucket. Separating out here allows me to run and rerun with srcBucket = `halconfig` while outputting to a temp dev bucket.

Reply:
Removed the `replaced` variable and will just rewrite the same content back to that Object.